web-components/src/components/combobox/ComboBox.ts

```diff
@@ -1324,86 +1441,97 @@ export namespace ComboBox {
-                        !(this.isOptGroup && !this.groupExpandedList.includes(this.getOptionGroupName(option)))
-                      ) {
-                        return html`
-                          <li
-                            id=${this.getOptionId(option)}
-                            title="${this.getOptionValue(option)}"
-                            part="combobox-option"
-                            role="option"
-                            class="md-combobox-option"
-                            aria-label=${this.getOptionValue(option)}
-                            aria-selected=${this.getAriaState(optionIndex)}
-                            tabindex="-1"
-                            @click=${this.handleListClick}
-                            aria-checked=${ifDefined(this.isMulti ? this.isOptionChecked(option) : undefined)}
-                          >
-                            ${this.isMulti
-                              ? html`
-                                  <span class="select-option">
-                                    <md-icon name="icon-check_14"></md-icon>
-                                  </span>
-                                `
-                              : nothing}
-                            <span part="label" class="select-label">
-                              ${this.isCustomContent
-                                ? this.getCustomContent(option)
-                                : findHighlight(
-                                    this.getOptionValue(option),
-                                    this.trimSpace ? this.inputValue.replace(/\s+/g, "") : this.inputValue
-                                  ).map(({ text, matching }) =>
-                                    matching
-                                      ? html`
-                                          <span class="highlight-text">${text}</span>
-                                        `
-                                      : html`
-                                          <span class="selected-label-text">${text}</span>
-                                        `
-                                  )}
-                            </span>
-                          </li>
-                        `;
-                      }
-                    }
-                  )}
+                  ${this.isMulti && this.allowSelectAll && this.expanded ? this.getSelectAllOption() : nothing}
+                  ${!this.checkForVirtualScroll()
+                    ? repeat(
+                        this.filterOptions(this.trimSpace ? this.inputValue.replace(/\s+/g, "") : this.inputValue),
+                        (option: string | OptionMember) => this.getOptionId(option),
+                        (option: string | OptionMember, optionIndex) => {
+                          if (typeof option !== "string" && this.isOptGroup && option.isLabel === "true") {
+                            return html`
+                              <div
```

Comment:
is it possible to extract seprate component ofr each if/else condition for better readability

Reply:
Fixed